src/skin/qml/qmlskin.cpp

```diff
@@ -204,6 +206,27 @@ QWidget* QmlSkin::loadSkin(QWidget* pParent,
                         return pConfigProxy;
                     }));
 
+    qmlRegisterUncreatableType<QmlLibraryTrackListModel>("Mixxx",
+            0,
+            1,
+            "LibraryTrackListModel",
+            "LibraryTrackListModel objects can't be created directly, "
+            "please use Mixxx.Library.model");
```

Comment:
Is there a specific reason why we are making it uncreatable?

Reply:
Yes. You need to pass the actual library tablemodel to the constructor, which is not possible in qml. We could implement lazy setting, but that is more complex, requires exposing the underlying library track model to and I'm unclear what the benefit would be.

Reply:
The benefit would be that it would be more aligned with how the models are usually done in QML, but because of the technical challenges you just described, it obviously doesn't make much sense. The current solution is fine for me.

Reply:
Please add a comment briefly explaining the technical challenges.

Reply:
cafb002a517f5b8752496c3f25f296a50470c2fb